homeassistant/components/vicare/climate.py

```diff
@@ -78,15 +73,22 @@
 }
 
 VICARE_TO_HA_PRESET_HEATING = {
-    VICARE_PROGRAM_COMFORT: PRESET_COMFORT,
-    VICARE_PROGRAM_ECO: PRESET_ECO,
-    VICARE_PROGRAM_NORMAL: PRESET_NONE,
+    HeatingProgram.ACTIVE: PRESET_NONE,
+    HeatingProgram.COMFORT: PRESET_COMFORT,
+    HeatingProgram.ECO: PRESET_ECO,
+    HeatingProgram.EXTERNAL: PRESET_NONE,
+    HeatingProgram.FIXED: PRESET_NONE,
+    HeatingProgram.FORCED: PRESET_NONE,
+    HeatingProgram.HOLIDAY: PRESET_AWAY,
+    HeatingProgram.NORMAL: PRESET_HOME,
+    HeatingProgram.REDUCED: PRESET_SLEEP,
+    HeatingProgram.STANDBY: PRESET_NONE,
```

Comment:
@fb22 with [all the new programs added](https://github.com/somm15/PyViCare/blob/7b9a3218eb54dacbada54dc59e6e4f85ea52c5ba/PyViCare/PyViCareHeatingDevice.py#L606C9-L606C20) how can the mapping then be?

Reply:
HA modes | ViCare modes
-- | --
heating | heating
cooling | cooling
heating/cooling | heating/cooling
off | standby
auto | -

HA actions | ViCare actions
-- | --
off | standby
heating | reduced, normal, comfort, reducedHeating, normalHeating, comfortHeating, eco, comfortEnergySaving, normalEnergySaving, reducedEnergySaving, summerEco
cooling | reducedCooling, normalCooling, comfortCooling, comfortCoolingEnergySaving, normalCoolingEnergySaving, reducedCoolingEnergySaving

HA presets | ViCare presets
-- | --
reduced | reduced, reducedCooling, reducedHeating
normal | normal, normalCooling, normalHeating
comfort | comfort, comfortCooling, comfortHeating
eco | eco, comfortCoolingEnergySaving, comfortEnergySaving, normalCoolingEnergySaving, normalEnergySaving, reducedCoolingEnergySaving, reducedEnergySaving, summerEco
holiday | holiday
home | holidayAtHome
external | external
fixed | fixed
manual | manual

Only **mode** is selectable. **Presets** should not be selectable because most presets are set in the timer programs and for holiday you need to enter a schedule. The active preset can be read with getActiveProgram.

**HVAC action** cannot be derived from getActiveProgram (e.g. if 'holiday' is set, getActiveProgram returns 'holiday' but you don't know if it's heating or cooling). The only possibility I see is to check the value of every heating/cooling program (one of them will return 'true'). A function is not yet available in PyViCare.

If 'forcedLastFromSchedule' is active, getActiveProgram will not return it. So I would not include this in the presets (maybe a binary sensor). Probably 'dhwPrecedence' and 'frostprotection' too.

I'm not sure if getActiveProgram will return an eco or EnergySaving program if it's active. Also 'external', 'fixed' and 'manual' need to be checked.